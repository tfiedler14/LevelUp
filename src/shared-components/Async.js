import * as React from 'react';
import { Text, View } from 'react-native';
import { connect } from 'react-redux';

export const Async = ({ render, loading }) => {
  return <View>{loading ? <Text>Loading...</Text> : render}</View>;
};

const mapStateToProps = state => {
  return {
    loading: state.loading
  };
};

export default connect(mapStateToProps)(Async);
