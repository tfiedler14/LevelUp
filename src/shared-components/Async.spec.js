import React from 'react';
import { shallow } from 'enzyme';
import { Async } from './Async';

describe('Async', () => {
  it('should render Async correctly', () => {
    const component = shallow(<Async render={{}} loading={false} />);

    expect(component).toMatchSnapshot();
  });
});
