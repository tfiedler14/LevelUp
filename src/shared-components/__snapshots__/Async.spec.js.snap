// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Async should render Async correctly 1`] = `
<View>
  <Component />
</View>
`;
